README.md
Build status: [![Build Status](https://travis-ci.com/JoinAmigo/amigo-web.svg?token=mqzQUzfs8sg4XGmA9aUn&branch=master)](https://travis-ci.com/JoinAmigo/amigo-web) on [master](https://github.com/JoinAmigo/amigo-web/tree/master)
[![Build Status](https://travis-ci.com/JoinAmigo/amigo-web.svg?token=mqzQUzfs8sg4XGmA9aUn&branch=master)](https://travis-ci.com/JoinAmigo/amigo-web) on [prod](https://github.com/JoinAmigo/amigo-web/tree/prod)

# Amigo API

[Amigo documentation](http://joinamigo.github.io/amigo-web/), refer [Update Documentation](https://github.com/JoinAmigo/amigo-web/#update-documentation) section below to learn how to update it.

## Getting up and running

###Verify xcode command line tools:

```xcode-select -p```

should show ```/Applications/Xcode.app/Contents/Developer```

###Install HomeBrew - The missing package manager for OS X:
 
   * Install homebrew from https://brew.sh by running the following command:
   * `ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"`
   * Verify homebrew is up to date
   * `brew update`
   * Verify homebrew has no issues
   * `brew doctor`

###Install required HomeBrew packages

```
brew bundle
```

should output success like:

```
Succeeded in tapping homebrew/bundle
Succeeded in installing openssl
Succeeded in installing python
Succeeded in installing postgresql
Succeeded in installing redis
Succeeded in installing phantomjs

Success: 6 Fail: 0
```

Note you might need to install Python via homebrew with open ssl

```
brew install python --with-brewed-openssl
```

###Install Python packages

```
pip install --upgrade pip setuptools
pip install fabric
pip install redis
pip install postgres
pip install virtualenv
pip install -r requirements/development.txt
```

###Prepare a local machine for development:

`fab init`

###Setup your server environment:

- Get the heroku config by running: `heroku config -s -a amigo-master`
- edit local `.env` file contaning project secrets, that are not part of the Source Code:
It should have these values:

```
DJANGO_SECRET_KEY="Secret_Key_Created_by_Fab_Init"
DATABASE_URL="postgres://localhost/amigo"
TWILIO_ACCOUNT_SID="ACc64a86f188d5f0218a524650aadfa6ac"
TWILIO_AUTH_TOKEN="b134efe9ec64ba8615e65887fa6f95ee"
TWILIO_PHONE_NUMBER="+18472644624"
USERVOICE_SUBDOMAIN_NAME="joinamigo"
USERVOICE_API_KEY="V0zo5bsSu55k1Hz8g127DA"
USERVOICE_API_SECRET="quUFMz9blvpRk0Jrh0Dp3QFo5iSxR73XMdArAfx37qw"
```
Add the following values:

```
DEBUG=True
DJANGO_DEBUG=True
DJANGO_ENVIRONMENT=[Dev]
DJANGO_CONFIGURATION=Development
ZEROPUSH_AUTH_TOKEN=Value_From_Heroku_Config
DJANGO_SETTINGS_MODULE=settings.development
DJANGO_SITE_DOMAIN=localhost:8000

```

###Start local server with all required services (redis,postgres)
```
fab honcho
```

#### To start services separately
Start Postgres

```
postgres -D /usr/local/var/postgres
```

Stop Postgres

`pg_ctl stop -D /usr/local/var/postgres -m fast`

Start redis

```
redis-server /usr/local/etc/redis.conf
```

To start local server run:

`fab serve` - start django server on [http://localhost:8000/](http://localhost:8000/)

To kill server:

`kill -9 $(lsof -ti tcp:8000)`

###To verify that cloned code run tests:

`fab test` - run the local server tests

###Create local administrator

`fab manage:createsuperuser` - create local admin for [http://localhost:8000/admin](http://localhost:8000/admin)

 - email: `alldev@amigo.io`
 - phone number: `+1 415-555-1111`
 - password: `amigo` insecure password for only local development server

###Development tools

####Load DJango development shell:

`fab shell`

####Update Documentation

`fab serve_docs`
This will run a local HTTP server and as you are editing \*.md files you can instantly view your changes at: `127.0.0.1:8001` (Read [mkdocs](http://www.mkdocs.org/) to learn more).

`fab deploy_docs`
This will update server documentation on github pages: [http://joinamigo.github.io/amigo-web/](http://joinamigo.github.io/amigo-web/):

####List available fabric tasks:

`fab list`

#### Heroku commands

Clone existing heroku app

```heroku git:clone -a amigo-master```

Show heroku configuration

```heroku config -s -a amigo-master```

Create Administrator:

`heroku run python manage.py createsuperuser --app amigo-master`

__TODO:__ find out why `heroku run fab createsuperuser --app amigo-master` does not work

Load `.env` variables to your current shell

```export $(cat .env | xargs)```

## Contributing

Golden Rule:

> Anything in **master** is always **deployable**.

Avoid working on `master` branch, create a new branch with meaningful name, send pull request asap. Be vocal!

Refer to [CONTRIBUTING.md][contributing]

[contributing]: http://github.com/JoinAmigo/amigo-web/tree/master/CONTRIBUTING.md

##Production deployment:
```shell
git checkout master
fab test
bumpversion release
bumpversion --no-tag patch # 'patch' can be replaced with 'minor' or 'major'
git push origin master
git push origin master --tags
git checkout prod
git rebase master
git push origin prod
```

For deploying to production:

Edit your local ` ~/.ssh/config` to have correct hostname from AWS:

```
Host web1.amigoapp.co
    HostName ec2-54-80-0-153.compute-1.amazonaws.com
    User ubuntu
    ForwardAgent yes
    Identityfile ~/.ssh/id_aws
```

Then run:

```shell
git checkout prod
git rebase master
fab prod configure
```
You may experience errors during inital setup because the environment variables are not set.  Just create a `.env` file in the `amigo-web folder`.  Also, if ngnix is not starting type `sudo service nginx restart` to reload your page with redirects in place.

When updating production ensure your static files are hosted in the [S3 com-amigo bucket](https://console.aws.amazon.com/s3/home?bucket=com-amigo), and they are made publicly accesssible, i.e. [https://com-amigo.s3.amazonaws.com/css/normalize.css](https://com-amigo.s3.amazonaws.com/css/normalize.css)

`gunicorn --pythonpath="$PWD" wsgi:application` - restart the django server
`kill -9 $(lsof -ti tcp:8000)` - kill a process on a port, such as port 8000.


docs/release_notes.md
# Release Notes

## [2.2.0]

__Date:__ 2016-08-16

__Added__

- brand new UX released
- implemented randomized sample sent/received events
- add `POST` `/users/register_fb_token` API
- add `GET` `/users/fb_friends` API

## [2.1.0-dev]

__Date:__ [Unreleased](https://github.com/JoinAmigo/amigo-web/issues/133)

__Added__

- `faded_photo` to `/events/timeline` invitees objects
- `GET /events/timeline` endpoint
- `GET`, `POST`, `DELETE` for `/users/favorite_contacts`
- `POST /auth/verify_pin`

__Changed__

- Update the response format of `POST /users/from_phone_numbers`
- `/auth` reponse data.
- Upgrade to Django 1.8. (#147)

__Fixed__
- handle the case where two people are able to accept invite even all the spots are filled. (#143)

__Removed__

- all `/phonebooks` endpoints
- `/users/invite_to_amigo` endpoint.


## [2.0.0]

__Date:__ [31st July 2015](https://github.com/JoinAmigo/amigo-web/issues/38)

__Added__

- Mixpanel integration
- API Call for Badge Count (#90)
- API: add/update users' notification preferences (#37)
- Notification: Total spots changed for a event (#70)
- Notification: One Spot Left for Event (#69)
- Notification: upcoming event reminder (#103)
- Implement logout functionality (#89)
- Save/update users' phonebook on backend (#81)
- Update event notification change to be sent only attending users (#85)
- Notification: "One of your contact joined amigo" (#76)
- `invitees_photos` property to `GET /events` (#54)
- `rsvp_message` property to `POST /events/:id/accept` and `POST /events/:id/unaccept` (#34)
- `rsvp_message` to `invitees` array of object at `GET /events/:id` (#36)
- Optional `address` and `location` field to event object (#33)
- Database [persistent connections], for faster database access.
- `my_rsvp_message` property to `GET /events` and `GET /events/:id` (#45)
- `created` property to invitees resource so it can be sorted properly (#51)
- Api versioning support (followed by drf upgrade)
- `rsvp_time` in invitees for event detail (#58)
- `POST /users/invite_to_amigo` for inviting friends to amigo (#57)
- Add `venue_name` property to Event resource (#67)

__Changed__

- Implement/update new push notifications and sms messages (#65)
- Make time part in `event_date` optional by introducing `is_time_specified` property. (#46)
- Change `i_am_attending` property at `/events` resource nullable (#35)
- Rename `/events/:id/unaccept` to `/events/:id/reject` (#44)
- Upgrade to drf 3.0 (#43)
- For users without images, set image to null (#47)

[2.2.0]: https://github.com/JoinAmigo/amigo-web/compare/v2.2.0...master
[2.1.0-dev]: https://github.com/JoinAmigo/amigo-web/compare/v2.0.0...master
[2.0.0]: https://github.com/JoinAmigo/amigo-web/compare/v1.5.0...v2.0.0


settings/common.py
# -*- coding: utf-8 -*-
'''Django settings for Amigo project.

see: https://docs.djangoproject.com/en/dev/ref/settings/
'''
# Standard Library
from os.path import dirname, join

# Third Party Stuff
import environ
from configurations import Configuration

# Build paths inside the project like this: os.path.join(ROOT_DIR, ...)
ROOT_DIR = dirname(dirname(__file__))
APP_DIR = join(ROOT_DIR, 'amigo')

env = environ.Env()


# Common Configurations
# ========================================================================
class Common(Configuration):
    # APP CONFIGURATION
    # -----------------
    INSTALLED_APPS = (
        'django.contrib.auth',
        'django.contrib.contenttypes',
        'django.contrib.sessions',
        'django_sites',  # http://niwibe.github.io/django-sites/
        'django.contrib.messages',
        'django.contrib.staticfiles',
        'django.contrib.humanize',
        'grappelli',
        'django.contrib.admin',

        'amigo.base',
        'amigo.users',
        'amigo.events',
        'amigo.pages',
        'amigo.notifications',
        'tz_detect',

        'django_extensions',  # http://django-extensions.readthedocs.org/
        'rest_framework',  # http://www.django-rest-framework.org/
        'django_twilio_sms',  # https://github.com/nigma/django-twilio-sms
        'zeropush',  # https://github.com/hakanw/django-zeropush
        'versatileimagefield',  # https://github.com/WGBH/django-versatileimagefield/
        'corsheaders',  # https://github.com/ottoyiu/django-cors-headers
        'airbrake',  # https://github.com/airbrake/airbrake-django
    )

    # MIDDLEWARE CONFIGURATION
    # Note: Order in which they are added are important
    MIDDLEWARE_CLASSES = (
        # Make sure djangosecure.middleware.SecurityMiddleware is the first
        # middleware class listed
        'djangosecure.middleware.SecurityMiddleware',
        'django.contrib.sessions.middleware.SessionMiddleware',
        'corsheaders.middleware.CorsMiddleware',
        'django.middleware.common.CommonMiddleware',
        'django.middleware.csrf.CsrfViewMiddleware',
        'django.contrib.auth.middleware.AuthenticationMiddleware',
        'django.contrib.messages.middleware.MessageMiddleware',
        'django.middleware.clickjacking.XFrameOptionsMiddleware',
        'tz_detect.middleware.TimezoneMiddleware',
    )

    # See: https://docs.djangoproject.com/en/dev/ref/settings/#debug
    # Defaults to false, which is safe, enable them only in development.
    DEBUG = env.bool('DJANGO_DEBUG', default=False)

    # See: https://docs.djangoproject.com/en/dev/ref/settings/#template-debug
    TEMPLATE_DEBUG = DEBUG

    # See:
    # https://docs.djangoproject.com/en/dev/ref/settings/#std:setting-FIXTURE_DIRS
    FIXTURE_DIRS = (
        join(APP_DIR, 'fixtures'),
    )

    # MANAGER CONFIGURATION
    # See: https://docs.djangoproject.com/en/dev/ref/settings/#admins
    ADMINS = (
        ('Amigo Admin', 'info@amigo.io'),
        ('Paul Zabelin', 'paul@amigo.io'),
        ('Dmitri Zdorov', 'dmitri@amigo.io'),
        ('Wei Ye', 'wei@amigo.io'),
    )

    # See: https://docs.djangoproject.com/en/dev/ref/settings/#managers
    MANAGERS = ADMINS
    # END MANAGER CONFIGURATION

    # EMAIL CONFIGURATION
    EMAIL_BACKEND = env('DJANGO_EMAIL_BACKEND',
                        default='django.core.mail.backends.smtp.EmailBackend')

    # END EMAIL CONFIGURATION

    # Local time zone for this installation. Choices can be found here:
    # http://en.wikipedia.org/wiki/List_of_tz_zones_by_name
    # although not all choices may be available on all operating systems.
    # In a Windows environment this must be set to your system time zone.
    TIME_ZONE = 'UTC'

    # Language code for this installation. All choices can be found here:
    # http://www.i18nguy.com/unicode/language-identifiers.html
    LANGUAGE_CODE = 'en-us'

    # If you set this to False, Django will make some optimizations so as not
    # to load the internationalization machinery.
    USE_I18N = True

    # If you set this to False, Django will not format dates, numbers and
    # calendars according to the current locale.
    USE_L10N = True

    # If you set this to False, Django will not use timezone-aware datetimes.
    USE_TZ = True

    # TEMPLATE CONFIGURATION
    TEMPLATE_CONTEXT_PROCESSORS = (
        'django.contrib.auth.context_processors.auth',
        'django.core.context_processors.debug',
        'django.core.context_processors.i18n',
        'django.core.context_processors.media',
        'django.core.context_processors.static',
        'django.core.context_processors.tz',
        'django.contrib.messages.context_processors.messages',
        'django.core.context_processors.request',
        # Your stuff: custom template context processers go here
    )

    # See: https://docs.djangoproject.com/en/dev/ref/settings/#template-dirs
    TEMPLATE_DIRS = (
        join(APP_DIR, 'templates'),
    )

    TEMPLATE_LOADERS = (
        'django.template.loaders.filesystem.Loader',
        'django.template.loaders.app_directories.Loader',
    )

    # STATIC FILE CONFIGURATION
    # See: https://docs.djangoproject.com/en/dev/ref/settings/#static-root
    STATIC_ROOT = join(ROOT_DIR, '.staticfiles')

    # See: https://docs.djangoproject.com/en/dev/ref/settings/#static-url
    STATIC_URL = '/static/'

    # See:
    # https://docs.djangoproject.com/en/dev/ref/contrib/staticfiles/#std:setting-STATICFILES_DIRS
    STATICFILES_DIRS = (
    )

    # List of finder classes that know how to find static files in
    # various locations.
    STATICFILES_FINDERS = (
        'django.contrib.staticfiles.finders.FileSystemFinder',
        'django.contrib.staticfiles.finders.AppDirectoriesFinder',
    )
    # END STATIC FILE CONFIGURATION

    # MEDIA CONFIGURATION
    # See: https://docs.djangoproject.com/en/dev/ref/settings/#media-root
    MEDIA_ROOT = join(ROOT_DIR, '.media')

    # See: https://docs.djangoproject.com/en/dev/ref/settings/#media-url
    MEDIA_URL = '/media/'
    # END MEDIA CONFIGURATION

    # URL Configuration
    ROOT_URLCONF = 'amigo.urls'

    # See: https://docs.djangoproject.com/en/dev/ref/settings/#wsgi-application
    WSGI_APPLICATION = 'wsgi.application'
    # End URL Configuration

    # AUTHENTICATION CONFIGURATION (only for django admin)
    AUTHENTICATION_BACKENDS = (
        "django.contrib.auth.backends.ModelBackend",
    )

    # CORS CONFIG
    CORS_ORIGIN_ALLOW_ALL = True

    # SLUGLIFIER
    AUTOSLUG_SLUGIFY_FUNCTION = "slugify.slugify"
    # END SLUGLIFIER

    # LOGGING CONFIGURATION
    # See: https://docs.djangoproject.com/en/dev/ref/settings/#logging
    # A sample logging configuration. The only tangible logging
    # performed by this configuration is to send an email to
    # the site admins on every HTTP 500 error when DEBUG=False.
    # See http://docs.djangoproject.com/en/dev/topics/logging for
    # more details on how to customize your logging configuration.
    LOGGING = {
        'version': 1,
        'disable_existing_loggers': False,
        'filters': {
            'require_debug_false': {
                '()': 'django.utils.log.RequireDebugFalse'
            }
        },
        'formatters': {
            'complete': {
                'format': '%(levelname)s:%(asctime)s:%(module)s %(message)s'
            },
            'simple': {
                'format': '%(levelname)s:%(asctime)s: %(message)s'
            },
            'null': {
                'format': '%(message)s',
            },
        },
        'handlers': {
            'null': {
                'level': 'DEBUG',
                'class': 'django.utils.log.NullHandler',
            },
            'console': {
                'level': 'DEBUG',
                'class': 'logging.StreamHandler',
                'formatter': 'simple',
            },
            'mail_admins': {
                'level': 'ERROR',
                'filters': ['require_debug_false'],
                'class': 'django.utils.log.AdminEmailHandler'
            },
        },
        'loggers': {
            'django': {
                'handlers': ['null'],
                'propagate': True,
                'level': 'INFO',
            },
            'django.request': {
                'handlers': ['mail_admins', 'console'],
                'level': 'ERROR',
                'propagate': True,
            },
        }
    }

    # DATABASE CONFIGURATION
    # ------------------------------------------------------------------------------
    # See: https://docs.djangoproject.com/en/dev/ref/settings/#databases
    DATABASES = {
        'default': env.db("DATABASE_URL", default="postgres://localhost/amigo"),
    }
    DATABASES['default']['ATOMIC_REQUESTS'] = True
    DATABASES['default']['CONN_MAX_AGE'] = 60

    SITES = {
        "local": {"domain": "localhost:8000", "scheme": "http", "name": "localhost"},
        "remote": {
            "domain": env('SITE_DOMAIN', default=''),
            "scheme": env('SITE_SCHEME', default='https'),
            "name": env('SITE_NAME', default='Amigo'),
        },
    }

    SITE_ID = "remote"

    # Django Rest Framework
    REST_FRAMEWORK = {
        'DEFAULT_PAGINATION_CLASS': 'amigo.base.api.pagination.PageNumberPagination',
        'PAGE_SIZE': 30,

        # 'Accept' header based versioning
        # http://www.django-rest-framework.org/api-guide/versioning/
        'DEFAULT_VERSIONING_CLASS': 'rest_framework.versioning.AcceptHeaderVersioning',
        'DEFAULT_VERSION': '1.0',
        'ALLOWED_VERSIONS': ['1.0', ],
        'VERSION_PARAMETER': 'version',


        # Use hyperlinked styles by default.
        # Only used if the `serializer_class` attribute is not set on a view.
        'DEFAULT_MODEL_SERIALIZER_CLASS': 'rest_framework.serializers.HyperlinkedModelSerializer',

        # Use Django's standard `django.contrib.auth` permissions,
        # or allow read-only access for unauthenticated users.
        'DEFAULT_PERMISSION_CLASSES': [
            'rest_framework.permissions.IsAuthenticated',
        ],
        'DEFAULT_AUTHENTICATION_CLASSES': (
            # Mainly used by amigo-ios
            "amigo.auth.backends.Token",

            # Mainly used for api debug.
            "amigo.auth.backends.Session",
        ),
        "EXCEPTION_HANDLER": "amigo.base.exceptions.exception_handler",
    }

    # MIGRATIONS CONFIGURATION
    # ------------------------------------------------------------------------------
    MIGRATION_MODULES = {
        'zeropush': 'amigo.zeropush.migrations'
    }

    # Django Rest Framework
    PUBLIC_REGISTER_ENABLED = env.bool('DJANGO_PUBLIC_REGISTER_ENABLED',
                                       default=True)

    USE_ETAGS = True

    # Authentication
    AUTH_USER_MODEL = "users.User"

    # Amigo
    AMIGO_APPSTORE_URL = env("AMIGO_APPSTORE_URL", default='http://amigo.io')
    AMIGO_APP_URL_SCHEME = env("AMIGO_APP_URL_SCHEME", default='amigo')
    HOURS_TO_EVENT_DEACTIVATATION = 12

    DEFAULT_AVATAR_SIZE = 60                # 60x60 pixels
    DEFAULT_BIG_AVATAR_SIZE = 250           # 250x250 pixels
    VERSATILEIMAGEFIELD_SETTINGS = {
        # The amount of time, in seconds, that references to created images
        # should be stored in the cache. Defaults to `2592000` (30 days)
        'cache_length': 2592000,
        # The name of the cache you'd like `django-versatileimagefield` to use.
        # Defaults to 'versatileimagefield_cache'. If no cache exists to the name
        # provided, the 'default' cache will be used.
        'cache_name': 'vif_cache',
        # The save quality of modified JPEG images. More info here:
        # http://pillow.readthedocs.org/en/latest/handbook/image-file-formats.html#jpeg
        # Defaults to 70
        'jpeg_resize_quality': 95,
        # A path on disc to an image that will be used as a 'placeholder'
        # for non-existant images.
        # If 'global_placeholder_image' is unset, the excellent, free-to-use
        # http://placehold.it service will be used instead.
        'global_placeholder_image': None,
        # The name of the top-level folder within your storage to save all
        # sized images. Defaults to '__sized__'
        'sized_directory_name': "_s_",
        # The name of the directory to save all filtered images within.
        # Defaults to '__filtered__':
        'filtered_directory_name': "_f_"
    }

    # TWILIO Configuration
    # see: https://www.twilio.com/user/account
    TWILIO_ACCOUNT_SID = env('TWILIO_ACCOUNT_SID', default='')
    TWILIO_AUTH_TOKEN = env('TWILIO_AUTH_TOKEN', default='')
    TWILIO_PHONE_NUMBER = env('TWILIO_PHONE_NUMBER', default='')

    TWILIO_CALLBACK_DOMAIN = env('TWILIO_CALLBACK_DOMAIN', default='')
    TWILIO_CALLBACK_USE_HTTPS = env('SITE_SCHEME', default="http")
    TWILIO_DRY_MODE = env('TWILIO_DRY_MODE', default=False)
    TWILIO_SKIP_SIGNATURE_VALIDATION = env('TWILIO_SKIP_SIGNATURE_VALIDATION',
                                           default=False)
    SMS_VALIDATION_LENGTH = 4
    SMS_VALIDATION_MESSAGE = ("Presto! {} is your secret code to verify"
                              " or tap this link to login: " + AMIGO_APP_URL_SCHEME + "://verify/{}")

    # Invite template
    SMS_INVITE_TEMPLATE = ("{user_name} is using Amigo to make it easier to get together with friends and "
                           "thinks you should join too! Get the app here: {app_store_link}")
    # -------- SMS configuration

    # ZERO PUSH
    ZEROPUSH_AUTH_TOKEN = env('ZEROPUSH_AUTH_TOKEN', default='')
    DISABLE_PUSH_NOTIFICATION = False
    ENVIRONMENT = ''

    CACHE_KEYS = {
        'event_invitee_photo_urls': 'event_i_p_urls:{event_id}'
    }

    # Analytics
    # -------------------------------------------------------------------------
    MIXPANEL_PROJECT_TOKEN = env('MIXPANEL_PROJECT_TOKEN', default='')

    # CACHING
    # ------------------------------------------------------------------------------
    CACHES = {
        'default': {
            'BACKEND': 'redis_lock.django_cache.RedisCache',
            "LOCATION": "{0}/{1}".format(env('REDISTOGO_URL', default="redis://localhost:6379").strip("/"), 0),
            'OPTIONS': {
                "CLIENT_CLASS": "django_redis.client.DefaultClient",
                # mimics memcache behavior.
                # http://niwinz.github.io/django-redis/latest/#_memcached_exceptions_behavior
                "IGNORE_EXCEPTIONS": env.bool('CACHE_IGNORE_EXCEPTIONS', default=False),
                'PARSER_CLASS': 'redis.connection.HiredisParser',
                'CONNECTION_POOL_CLASS': 'redis.BlockingConnectionPool',
                'CONNECTION_POOL_CLASS_KWARGS': {
                    'max_connections': 50,
                    'timeout': 20,
                },
            }
        }
    }

    # Most Common Time Zones
    TZ_DETECT_COUNTRIES = ('CN', 'US', 'IN', 'JP', 'BR', 'RU', 'DE', 'FR', 'GB')

    GRAPPELLI_ADMIN_TITLE = "Amigo Administration"

    # Airbrake settings
    AIRBRAKE = {
        'API_KEY': 'b57181ddbc5fee66b42ee8b16fc39d61',
        'API_URL': 'https://airbrake.io/',
        'USE_SSL': True,
        'TIMEOUT': 5,
        'ENVIRONMENT': 'production',
    }

    # Uservoice settings
    USERVOICE_SUBDOMAIN_NAME = env('USERVOICE_SUBDOMAIN_NAME', default='')
    USERVOICE_API_KEY = env('USERVOICE_API_KEY', default='')
    USERVOICE_API_SECRET = env('USERVOICE_API_SECRET', default='')
    USERVOICE_SSO_KEY = env('USERVOICE_SSO_KEY', default='')
    USERVOICE_URL = env('USERVOICE_URL', default='')
